France-ioi/AlgoreaFrontend
# fix user links

## Done

- Retreat all user links:
  - left nav tree
  - breadcrumbs
  - profile link
- allow to give user routes to `urlArrayForGroupRoute` and `group-router`
- create `userLink` pipe (already created in another [PR to be merged](https://github.com/France-ioi/AlgoreaFrontend/pull/627))

## Tests

This is a refactoring PR, therefore the production and branch apps should behave exactly the same

- [x] Breadcrumbs: [production app](https://dev.algorea.org/en/#/groups/users/670968966872011405) / [branch app](https://dev.algorea.org/branch/fix/user-links/en/#/groups/users/670968966872011405)
  1. Click on the breadcrumb
  2. You should have navigated to the progress page
- [x] Profile link: [production app](https://dev.algorea.org/en/#/groups/by-id/7650735426882403033;path=52767158366271444,672913018859223173/details) / [branch app](https://dev.algorea.org/branch/fix/user-links/en/#/groups/by-id/7650735426882403033;path=52767158366271444,672913018859223173/details)
  1. Click on the user login at the top right of the screen
  2. Select option "profile"
  3. You should be redirected to the user page "personal datas"
- [x] left nav tree for group links: [production app](https://dev.algorea.org/branch/fix/user-links/en/#/groups/by-id/52767158366271444;path=/details) / [branch app](https://dev.algorea.org/en/#/groups/by-id/52767158366271444;path=/details)
  1. Click on a subgroup
  2. You should have navigated to the subgroup, the path should have been provided
- [ ] left nav tree for user links:
  - I did not find any case where the node was a user.
